blog/2025-02-19-elastic-components.md

```diff
@@ -0,0 +1,520 @@
+also genuinely benefit from true pay-as-you-go characteristics. Consequently, more cloud providers are converging
+towards this new architectural paradigm.
+
+The core capability of "flexible configurability" in Serverless technology focuses on enabling specific cloud usage
+scenarios to fully utilize elastic resources through simple, minimally invasive, and highly configurable methods. Its
+essence lies in resolving the conflict between capacity planning and actual cluster load configuration. This article
+will sequentially introduce two configurable components — `WorkloadSpread` and `UnitedDeployment` — discussing their
+core capabilities, technical principles, advantages and disadvantages, as well as real-world applications. Through these
+discussions, we aim to share OpenKruise's technical evolution and considerations in addressing Serverless workload
+elasticity.
+
+# Overview of Elastic Scenarios
+
+As Serverless technology matures, more enterprises prefer using elastic resources (such as Alibaba Cloud ACS Serverless
+container instances) over static resources (like managed resource pools or self-built IDC data centers) to host
+applications with temporary, tidal, or bursty characteristics. This approach enhances resource utilization efficiency
+and reduces overall costs by adopting a pay-as-you-go model. Below are some typical elastic scenarios:
+
+1. Prioritize using self-built resources in offline IDC data centers; expand application replicas to the cloud when
+   resources are insufficient.
+2. Prefer using reserved managed node resource pools; use pay-as-you-go Serverless instances for additional replicas
+   when resources are insufficient.
+3. Use high-quality stable compute power (e.g., dedicated cloud server instances) first; then use lower-quality compute
+   power (e.g., Spot instances).
+4. Configure different resource quantities for container replicas deployed on different compute platforms (e.g., X86,
+   ARM, Serverless instances) to achieve similar performance.
+5. Inject different middleware configurations into replicas deployed on nodes versus Serverless environments (e.g.,
+   shared Daemon on nodes, Sidecar injection on Serverless).
+
+These components introduced in this article offer distinct advantages in solving the above problems. Users can choose
+appropriate capabilities based on their specific scenarios to effectively leverage elastic compute power.
+
+# Capabilities and Advantageous Scenarios of Two Components
+
+- **WorkloadSpread**: Utilizes a Mutating Webhook to intercept Pod creation requests that meet certain criteria and
+  apply Patch operations to inject differentiated configurations. Suitable for existing applications requiring multiple
+  elastic partitions with customized Pod Metadata and Spec fields.
+- **UnitedDeployment**: A natively supported workload for elastic partitioning and Pod customization, offering stronger
```

Comment:
A natively supported workload for elastic partitioning and Pod customization -> a workload with built-in capability of elastic partitioning and pod customization